File: 프로그래머스/lv1/12943. 콜라츠 추측/콜라츠 추측.py
def solution(num):
    answer = 0
    while True:
        if num == 1:
            break
        
        if num % 2 == 0:
            num = num / 2
        else:
            num = num * 3 + 1
        
        answer += 1
        if answer == 501:
            answer = -1
            break
    return answer

File: 프로그래머스/1/12943. 콜라츠 추측/콜라츠 추측.js
const isEven = (num) => num % 2 === 0 ? true : false

function solution(num) {
    var answer = 0;
    for (let i = 0; i < 500; i++) {
        if (num === 1) break
        
        if (isEven(num)) {
            num /= 2
        } else {
            num *= 3
            num += 1
        }
        
        answer++
        
 
    }
    return num === 1 ? answer : -1;
}